tests/unit/test_authz_model.py

```diff
@@ -805,3 +732,30 @@ def test_remove_agents_from_group():
         "agent_roles": ["test_role"],
         "rpc_capabilities": ["id.rpc1"]
     }
+
+
+def test_create_protected_topic():
+    ################
+    # Pre-requisites
+    ################
+    authz_map = authz.VolttronAuthzMap()
+
+    # Test empty authz
+    assert not authz_map.is_protected_topic(topic_name_pattern="foo")
+
+    # If _is_regex then use it,
+    # else topic_prefix.*
+    non_regex_topic = "devices/building1"
+    assert authz_map.create_protected_topics(topic_name_patterns=[non_regex_topic])
+
+    # prefix match
+    assert authz_map.is_protected_topic(topic_name_pattern=non_regex_topic)
+    assert authz_map.is_protected_topic(topic_name_pattern=non_regex_topic + "false")
+    assert not authz_map.is_protected_topic(topic_name_pattern=non_regex_topic[:-1])
+
+    reg_ex_topic = "/devices/building2/*/"
```

Comment:
shouldn't the topic pattern be ".*" to be valid?

Reply:
This is valid it works out just as it is and follows the correct chain of matches.